tunnel run fix

Hilimii opened the pull request:
Adds a new bool, and logic to the `onSplit ` event to stop tunnel run split from triggering twice.
Added some comments here and there.

Tested on my end and working.

Hilimii committed:
tunnel run fix

Adds a new bool, and logic to the `onSplit` event to stop tunnel run split from triggering twice.
Added some comments here and there.

jwfraustro commented on PW_Autosplitter.asl:
```diff
@@ -184,26 +188,44 @@ init
+{
+    // Tunnel run logic for culling multiple splits
+        // Sets beatTunnel to true when the exact criteria for a tunnel split are met
+    if
+    (
+        current.airUnitArrayLength == ( old.airUnitArrayLength + 4)
+        &&
+        vars.GetLevelID() == "mf_04" && settings["TunnelRun"] == true && vars.beatTunnel == false
+    )
+    {
+        vars.beatTunnel = true;
+    }
```
It might be easier to just move these checks into the main `split` check. i.e.
```
else if (vars.GetLevelID() == "mf_04" && settings["TunnelRun"] == true && vars.beatTunnel == false){
    if (current.airUnitArrayLength == ( old.airUnitArrayLength + 4)){
      vars.beatTunnel = true;
      return true;
    }
}
```

You'll have to check my syntax though, I'm just typing this out in the GH editor

Hilimii replied:
Ahhhhh thank you for this suggestion, I tried this earlier and wondered why it didn't work. Nesting another 'if' helped to solve the issue.

jwfraustro commented on PW_Autosplitter.asl:
```diff
@@ -184,26 +188,44 @@ init
 split
 {
-    // Trigger a split when missionComplete transitions from 2 to 3 (for most missions) or at the end of Kings or Faust.
+    // Trigger a split when missionComplete transitions from 2 to 3 (for most missions) or at the end of Kings or Faust. Redirects to functions in init.
```
I'd probably say `Call functions in init`.

I like the addition of these comments.

Hilimii replied:
i updated this to:
"// Calls relevant functions in 'init' to trigger splits for Kings and Faust."

Hilimii committed:
Revision for more efficient code

Sets beatTunnel to true within the split event itself. Removed onSplit event.
Changed some comments for clarity.

Hilimii commented:
I've implemented your requested changes :)

I think the entire tunnel run split logic could be moved into its own function in the future to make it a bit cleaner tbh, but that's a job for the future. Priority is to fix it atm.

jwfraustro commented:
> I've implemented your requested changes :)
> 
> 
> 
> I think the entire tunnel run split logic could be moved into its own function in the future to make it a bit cleaner tbh, but that's a job for the future. Priority is to fix it atm.

Excellent, I'm glad it worked out. I agree it could probably be split off, but it's not a huge burden atm. If something else comes up with the tunnel run we should definitely do so at that time. 

I'll test it first thing in the morning (my time, ~14UTC)

jwfraustro commented:
I am having a little trouble getting the start trigger to fire. Tracking it down now.

jwfraustro commented:
Okay, it seemed to be on my end. I can confirm the split only fires once. Interestingly, the mission complete trigger still fires if you kill all the enemies at the end.

That might be an unintended feature though, I suppose. Means people can run both categories at once! Thanks for the work here!

Hilimii commented:
It's not a bug, it a feature 😉 Thank you for checking, merge on the way!
